pkg/apis/cluster/transport/enroundtripper.go

```diff
@@ -0,0 +1,30 @@
+package multicluster
+
+import "net/http"
+
+var _ EnhanceClusterGatewayRoundTripper = &enhanceClusterGatewayRoundTripper{}
+
+type EnhanceClusterGatewayRoundTripper interface {
```

Comment:
"enhance" is an ambiguity name. in favor of a name will give users an intuitive sense of what the utilitity is doing. fwiw, "ProxyPathPrependingClusterGatewayRoundTripper" might be better?